simtools/io_handler.py

```diff
@@ -50,151 +51,29 @@ def getOutputDirectory(filesLocation, label, mode=None):
-        Instance label.
-
-    Returns
-    -------
-    Path
-    """
-    return getOutputDirectory(filesLocation, label, "layout")
-
-
-def getRayTracingOutputDirectory(filesLocation, label):
-    """
-    Get output directory for ray tracing related files.
-
-    Parameters
-    ----------
-    filesLocation: str, or Path
-        Main location of the output files.
-    label: str
-        Instance label.
-
-    Returns
-    -------
-    Path
-    """
-    return getOutputDirectory(filesLocation, label, "ray-tracing")
-
-
-def getCorsikaOutputDirectory(filesLocation, label):
-    """
-    Get output directory for CORSIKA related files.
-
-    Parameters
-    ----------
-    filesLocation: str, or Path
-        Main location of the output files.
-    label: str
-        Instance label.
-
-    Returns
-    -------
-    Path
-    """
-    return getOutputDirectory(filesLocation, label, "corsika")
-
-
-def getCameraEfficiencyOutputDirectory(filesLocation, label):
-    """
-    Get output directory for camera efficiency related files.
-
-    Parameters
-    ----------
-    filesLocation: str, or Path
-        Main location of the output files.
-    label: str
-        Instance label.
-
-    Returns
-    -------
-    Path
-    """
-    return getOutputDirectory(filesLocation, label, "camera-efficiency")
-
-
-def getApplicationOutputDirectory(filesLocation, label):
-    """
-    Get output directory for applications related files.
-
-    Parameters
-    ----------
-    filesLocation: str, or Path
-        Main location of the output files.
-    label: str
-        Instance label.
-
-    Returns
-    -------
-    Path
-    """
-    return getOutputDirectory(filesLocation, label, "application-plots")
-
-
-def getArraySimulatorOutputDirectory(filesLocation, label):
-    """
-    Get output directory for array-simulator related files.
-
-    Parameters
-    ----------
-    filesLocation: str, or Path
-        Main location of the output files.
-    label: str
-        Instance label.
-
-    Returns
-    -------
-    Path
-    """
-    return getOutputDirectory(filesLocation, label, "array-simulator")
-
-
-def getDerivedOutputDirectory(filesLocation, label):
+def getOutputFile(fileName, label=None, dirType=None, test=False):
     """
-    Get output directory for derived values files.
+    Get path of an output file.
 
     Parameters
     ----------
-    filesLocation: str, or Path
-        Main location of the output files.
+    filesName: str
+        File name.
     label: str
         Instance label.
+    dirType: str
+        Name of the subdirectory (ray-tracing, model etc)
+    test: bool
+        If true, return test output location
 
     Returns
     -------
     Path
     """
-    return getOutputDirectory(filesLocation, label, "derived")
+    return getOutputDirectory(label=label, dirType=dirType, test=test).joinpath(fileName).absolute()
 
 
-def getDataFile(parentDir, fileName):
+def getDataFile(parentDir=None, fileName=None, test=False):
```

Comment:
Yes, a DataFile is input. From a practical point of view I also didn't want to add a few additional `ifs` to the original function to use `Path("tests/resources/")` or `cfg.get("dataLocation")` in case it is data.

Reply:
Would it be better to call it InputDataFile (or something similar)? That would make things clearer when reading the code.

Reply:
Sure, done!